### PR TITLE
Adds option to evaluate node_modules directory on page load

### PR DESCRIPTION
**Pull request by itsthatguy**
Addresses #14

### CODE REVIEW

**Comment by itsthatguy** [comment]
@cymen: ping

**Comment by cymen** [comment]
@itsthatguy Sorry, I've been on vacation. This looks good to me. I want to try it where I'm using it to check if there is a slowdown or not. Have you noticed anything or does it seem fine?

**Comment by itsthatguy** [comment]
@cymen I don't notice any obvious slowdown, on a project with 18 node packages... that doesn't mean there isn't one, but i haven't noticed it

**Comment by cymen** [comment]
@itsthatguy Okay, sounds good -- I want to try this today/tomorrow and then merge it in.

**Comment by itsthatguy** [comment]
@cymen :+1:

File: test/compilation_test.rb [hunk 1]
```diff
@@ -33,9 +35,52 @@ def copy_example_file(filename)
...
+
+    expected_output = fixture("application.out.js")
 
-    get "/assets/foo.js"
+    get "/assets/application.js"
+    assert_response :success
+    assert_equal expected_output, @response.body.strip
+  end
+
+  test "asset pipeline should not regenerate application.js when node_modules changes" do
```
Comment (steveklebanoff) [inline test/compilation_test.rb]: I'm somewhat new to browserify and browserify-rails, so I may be missing something, but: is this test name accurate?  From what I can tell, it looks like application.js **is** regenerated in this test.

Comment (itsthatguy) [inline test/compilation_test.rb]: `application.node_test_package_changed.out.js` is a duplication of `application.js` with the node-test-package without changes.

What the test checks is:
* node package changes
* the post-evaluate version of `application.js` matches the `application.node_test_package_changes.out.js` to ensure there are no changes

Comment (itsthatguy) [inline test/compilation_test.rb]: You're right, the test names for the 2 conditions were reversed. Great catch!

**Comment by cymen** [comment]
@itsthatguy Sorry, I got sick :(. I haven't forgotten and I'm better now so will review shortly.

**Comment by itsthatguy** [comment]
@cymen feel better man!

**Comment by cymen** [comment]
@itsthatguy Would you object to changing it so that it was off by default (same behavior as before) but can be turned on? If so, it would make merging faster (and later on it might make sense to change the default to on).

**Comment by itsthatguy** [comment]
No objection

**Comment by cymen** [comment]
@itsthatguy Do you mind pushing it up with it defaulting off? If you prefer, I can do it too (just didn't want to do it unless you were). Then I'll merge. I really didn't want to keep this from getting merged for so long and I apologize for that.

**Comment by itsthatguy** [comment]
Either way is fine, it will take me a few to make that change. If you can do it now, go ahead.

**Comment by cymen** [comment]
@itsthatguy cool, I'll do it now

**Comment by cymen** [comment]
@itsthatguy Oh wait, I can't without too much fiddling around (think it would be different PR?) -- I'll leave it to you. Sorry for confusion.

**Comment by itsthatguy** [comment]
I've invited you to the repo team, you should be able to push to that branch directly -- appending the PR.

**Comment by cymen** [comment]
@itsthatguy Nice, I'll go do that and hopefully be merging shortly!

**Comment by itsthatguy** [comment]
:+1:

**Comment by cymen** [comment]
@itsthatguy merged!

**Comment by cymen** [comment]
@itsthatguy I bumped the version to 0.6.0 w/ this and pushed the gem

**Comment by itsthatguy** [comment]
looks great, thanks @cymen!

**Comment by itsthatguy** [comment]
@cymen should we close #14?

**Comment by cymen** [comment]
@itsthatguy thanks for following up on #14 -- I was going to go check but see everything is :+1:

**Pull request merged**